app/src/main/java/org/oppia/app/parser/StringToFractionParser.kt

```diff
@@ -53,4 +102,18 @@ class StringToFractionParser {
   }
 
   private fun isInputNegative(inputText: String): Boolean = inputText.startsWith("-")
+
+  /** Enum to store the errors of [FractionInputInteractionView]. */
+  enum class FractionParsingError(@StringRes private var error: Int?) {
+    VALID(error = null),
+    INVALID_CHARS(error = R.string.fraction_error_invalid_chars),
+    INVALID_FORMAT(error = R.string.fraction_error_invalid_format),
+    DIVISION_BY_ZERO(error = R.string.fraction_error_divide_by_zero),
+    LARGER_THAN_SEVEN_DIGITS(error = R.string.fraction_error_larger_than_seven_digits);
```

Comment:
Maybe instead: NUMBER_TOO_LONG? Either seems fine to me.

Reply:
Refactored to NUMBER_TOO_LONG